src/main/java/org/testeditor/fixture/web/json/BrowserSettingsManager.java

```diff
@@ -43,33 +46,112 @@
         
         BrowserSetupReader reader = new BrowserSetupReader();
         List<BrowserSetupElement> elements = reader.readElements(FILE_NAME);
-
+        
         Platform currentPlattform = Platform.getCurrent();
+        checkRedundantElements(elements,currentPlattform);
         return getBrowserSpecificSetting(elements, currentPlattform);
-
     }
     
+    /**
+     * This method is for testing purposes.
+     * @param fileName 
+     *          of JSON test file for browserSetup.json.
+     * @return 
+     *          A List of BrowserSetupElement
+     */
+    protected static List<BrowserSetupElement> getBrowserSettings(String fileName) {
+        
+        BrowserSetupReader reader = new BrowserSetupReader();
+        List<BrowserSetupElement> elements = reader.readElements(fileName);
+        
+        Platform currentPlattform = Platform.getCurrent();
+        checkRedundantElements(elements,currentPlattform);
+        return getBrowserSpecificSetting(elements, currentPlattform);
+    }
+    
+    protected static void checkRedundantElements(List<BrowserSetupElement> elements, Platform platform) {
```

Comment:
I still don't get why you don't check duplicate entries, when actually adding them. If you would hold the BrowserSettings in a Map instead of BrowserSetting you could use a lot of convenience functions provided by the collections of java.